grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -827,14 +827,18 @@ class RequisitionService {
         return new Event(eventDate: eventDate, eventType: eventType, eventLocation: eventLocation, createdBy: currentUser)
     }
 
-    Requisition transitionRequisitionStatus(Requisition requisition, RequisitionStatus requisitionStatus, EventCode eventCode, User currentUser) {
+    Requisition transitionRequisitionStatus(Requisition requisition, RequisitionStatus requisitionStatus, EventCode eventCode, User currentUser, Comment comment = null) {
         requisition.status = requisitionStatus
         Event event = createEvent(eventCode, requisition.origin, new Date(), currentUser)
         requisition.addToEvents(event)
-        requisition.save()
+
+        if (comment) {
+            event.comment = comment
```

Comment:
I still kinda don't get why the same comment has to be in both event and requisition - I don't see it (`event.comment` being used anywhere), but I won't delay it any longer, but I hope to get some explanation anyway to understand the context.